File: typescript/src/code-smells/bloaters/data-clump.ts
export class Invoice {
  private readonly customerName: string
  private readonly customerCity: string
  private readonly customerStreet: string
  private readonly customerZip: string

  constructor(
    customerName: string,
    customerStreet: string,
    customerCity: string,
    customerZip: string,
  ) {
    this.customerZip = customerZip
    this.customerStreet = customerStreet
    this.customerCity = customerCity
    this.customerName = customerName
  }

  print(): string {
    return (
      `Factura para: ${this.customerName}\n` +
      `Dirección: ${this.customerStreet}, ${this.customerCity}, ${this.customerZip}`
    )
  }
}

export class ShippingLabel {
  private readonly customerName: string
  private readonly customerStreet: string
  private readonly customerCity: string
  private readonly customerZip: string

  constructor(
    customerName: string,
    customerStreet: string,
    customerCity: string,
    customerZip: string,
  ) {
    this.customerZip = customerZip
    this.customerCity = customerCity
    this.customerStreet = customerStreet
    this.customerName = customerName
  }

  print(): string {
    return (
      `Enviar a: ${this.customerName}\n` +
      `${this.customerStreet}, ${this.customerCity}, ${this.customerZip}`
    )
  }
}

File: python/src/code_smells/bloaters/data_clump.py
class Invoice:
    def __init__(
        self,
        customer_name: str,
        customer_street: str,
        customer_city: str,
        customer_zip: str,
    ):
        self.customer_zip = customer_zip
        self.customer_street = customer_street
        self.customer_city = customer_city
        self.customer_name = customer_name

    def print(self) -> str:
        return (
            f"Factura para: {self.customer_name}\n"
            f"Dirección: {self.customer_street}, {self.customer_city}, {self.customer_zip}"
        )


class ShippingLabel:
    def __init__(
        self,
        customer_name: str,
        customer_street: str,
        customer_city: str,
        customer_zip: str,
    ):
        self.customer_zip = customer_zip
        self.customer_city = customer_city
        self.customer_street = customer_street
        self.customer_name = customer_name

    def print(self) -> str:
        return (
            f"Enviar a: {self.customer_name}\n"
            f"{self.customer_street}, {self.customer_city}, {self.customer_zip}"
        )
